src/infra/core/src/transform_service_impl.rs

```diff
@@ -697,9 +705,39 @@ impl TransformServiceImpl {
         // TODO: Inject time source
         let next_operation = self
             .get_next_operation(&dataset_handle, self.time_source.now())
-            .await?;
+            .await;
+
+        if options.reset_derivatives_on_diverged_input
+            && let Err(transform_err) = &next_operation
+            && let TransformError::InvalidInterval(_) = transform_err
+        {
+            let compacting_result = self
+                .compacting_svc
+                .compact_dataset(
+                    &dataset_handle,
+                    CompactingOptions {
+                        keep_metadata_only: true,
+                        ..Default::default()
+                    },
+                    None,
+                )
+                .await
+                .int_err()?;
+
+            if let CompactingResult::Success { .. } = compacting_result {
```

Comment:
What if result is not succesful?

Reply:
If no we will return error as it is on the next line 
```if let Some(operation) = next_operation?```